lib/engine/game/base.rb

```diff
@@ -622,14 +620,19 @@ def self.filtered_actions(actions)
       end
 
       # Initialize actions respecting the undo state
-      def initialize_actions(actions)
+      def initialize_actions(actions, at_action: nil)
         @loading = true unless @strict
-
         filtered_actions, active_undos = self.class.filtered_actions(actions)
+
+        # Store all actions for history navigation
+        @raw_all_actions = actions
```

Comment:
what's the difference between raw_actions and this

Reply:
raw_actions is all the actions up to the current game state. For example, if the player is reviewing the game history, it will only include all actions to the point at which they are in the game history. Conversely, raw_all_actions is all the actions taken, even future ones. 

A lot of logic depends on raw_actions, so I created a new array to not break that logic.